File: src/types/mfa.ts
export interface TOTPConfig {
  secret: string;
  uri: string;
  qrCode: string;
  backupCodes: string[];
  algorithm: string;
  digits: number;
  step: number;
}

export interface MFAStatus {
  enabled: boolean;
  verified: boolean;
  lastVerified: string | null;
  method: 'totp' | 'backup' | null;
  recoveryCodesRemaining: number;
  lastFailedAttempt?: string;
  failedAttempts: number;
  blockedUntil?: string;
}

export interface MFAVerificationResult {
  success: boolean;
  method?: 'totp' | 'backup';
  timestamp?: string;
  error?: string;
  remainingAttempts?: number;
  blockDuration?: number;
}

export interface MFASettings {
  userId: string;
  secret: string;
  backupCodes: string[];
  lastRotated: string;
  algorithm: string;
  digits: number;
  step: number;
}

export interface MFAChallenge {
  challengeId: string;
  method: 'totp' | 'backup';
  timestamp: string;
  expiresAt: string;
  attempts: number;
  maxAttempts: number;
} 

File: src/types/mfa.py
from dataclasses import dataclass
from typing import Literal, Optional, List


@dataclass
class TOTPConfig:
    secret: str
    uri: str
    qrCode: str
    backupCodes: List[str]
    algorithm: str
    digits: int
    step: int


@dataclass
class MFAStatus:
    enabled: bool
    verified: bool
    lastVerified: Optional[str] = None
    method: Optional[Literal['totp', 'backup']] = None
    recoveryCodesRemaining: int
    lastFailedAttempt: Optional[str] = None
    failedAttempts: int
    blockedUntil: Optional[str] = None


@dataclass
class MFAVerificationResult:
    success: bool
    method: Optional[Literal['totp', 'backup']] = None
    timestamp: Optional[str] = None
    error: Optional[str] = None
    remainingAttempts: Optional[int] = None
    blockDuration: Optional[int] = None


@dataclass
class MFASettings:
    userId: str
    secret: str
    backupCodes: List[str]
    lastRotated: str
    algorithm: str
    digits: int
    step: int


@dataclass
class MFAChallenge:
    challengeId: str
    method: Literal['totp', 'backup']
    timestamp: str
    expiresAt: str
    attempts: int
    maxAttempts: int